crates/solidity/outputs/cargo/slang_solidity/src/generated/bindings/generated/binding_rules.rs

```diff
@@ -740,6 +918,76 @@ attribute symbol_reference = symbol  => type = "push_symbol", symbol = symbol, i
 }
 
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Event definitions
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+@event [EventDefinition @name name: [Identifier]] {
+  node def
+  attr (def) node_definition = @name
+  attr (def) definiens_node = @event
+
+  edge @event.def -> def
+}
+
+;; Make the event available to the enclosing contract/interface/library.
+;; NB. top-level events (ie. those defined at the file's level) are already
+;; covered above
+
+@contract [ContractDefinition members: [ContractMembers
+    item: [ContractMember @event variant: [EventDefinition]]
+]] {
+  edge @contract.type_members -> @event.def
+}
+
+@interface [InterfaceDefinition members: [InterfaceMembers
+    item: [ContractMember @event variant: [EventDefinition]]
+]] {
+  edge @interface.type_members -> @event.def
+}
+
+@library [LibraryDefinition members: [LibraryMembers
```

Comment:
Same question here: what do you think about putting `LibraryMembers` with the `LibraryDefinition` itself? Same for interfaces, contracts, etc...
I'm reading through most of this doc, where the parents section usually has the links/edges with its children, but here they are the opposite.
Thoughts? would that make it more readable?

Reply:
Yep, now that I take a closer look, this doesn't even make sense for preserving strict evaluation. I'm guessing at some point it was. I'll try to deduplicate all these rules as well since they all have the same structure (eg. linking members to their enclosing contract/library/interface)